### PR TITLE
Fix test and parser for `GenericHttpCredentials` as in #21281.

### PR DESCRIPTION
**Pull request by briantopping**

### CODE REVIEW

**Comment by johanandren** [comment]
Refs #21281

**Comment by johanandren** [comment]
OK TO TEST

File: akka-http-core/src/test/scala/akka/http/impl/model/parser/HttpHeaderSpec.scala [hunk 1]
```diff
@@ -138,8 +138,8 @@ class HttpHeaderSpec extends FreeSpec with Matchers {
       """Authorization: Fancy yes="n:o", nonce=42""" =!=
         Authorization(GenericHttpCredentials("Fancy", Map("yes" → "n:o", "nonce" → "42"))).renderedTo(
           """Fancy yes="n:o",nonce=42""")
-      """Authorization: Fancy yes=no,nonce="4\\2"""" =!=
-        Authorization(GenericHttpCredentials("Fancy", Map("yes" → "no", "nonce" → """4\2""")))
+      """Authorization: Fancy yes=no,foo=,nonce="4\\2"""" =!=
+        Authorization(GenericHttpCredentials("Fancy", Map("yes" → "no", "foo" → "", "nonce" → """4\2""")))
```
Comment (johanandren) [inline akka-http-core/src/test/scala/akka/http/impl/model/parser/HttpHeaderSpec.scala]: Cover empty quoted-string as well?

Comment (briantopping) [inline akka-http-core/src/test/scala/akka/http/impl/model/parser/HttpHeaderSpec.scala]: I don't think it does a good job of this, discussion in #21281.

**Comment by briantopping** [comment]
Defer to #21384

**Comment by jrudolph** [comment]
Thanks again, @briantopping for finding the bug (even if it was another one than originally thought...) :)

**Comment by briantopping** [comment]
Peer review is important to getting it right, I think I developed a cognitive bias toward behavior I saw over several million transactions on a local client system. Can't have this stuff floating away from the spec in small ways or it turns into a huge mess!

**Pull request closed without merging**